### ISSUE TITLE
JetBrains plugin next-tab/previous-tab bug(s)

### ISSUE THREAD

**Issue by eostrom** [wallabyjs/public#2565]
### Issue description or question

I'm using the Wallaby plugin in RubyMine, where it appears in the Run tab, alongside other processes:

![2020-11-12 at 7 51 PM](https://user-images.githubusercontent.com/8683/99018810-97bf2600-2520-11eb-9cb4-f9d3437e637c.png)

When Wallaby isn't active, the command-} and command-{ keys (on Mac) move right and left between Run processes – `dev`, `cy:start`, `lint` in this screenshot.

When Wallaby is active, I usually have its Failing Tests tab open, as shown. If I start on the `dev` tab and I press command-} five times, these things happen in order:

1. The Run window switches to the Wallaby Jest tab, as I'd expect.
2. *Within* the Wallaby Jest tab, it switches to Wallaby Console. I'm a little surprised by this, but I can see a case for it.
3. It switches to Value Explorer | Debugger, showing Value Explorer. Same.
4. _Within_ Value Explorer | Debugger, it switches from Value Explorer to Debugger. OK...
5. Nothing happens!

![2020-11-12 at 8 29 PM](https://user-images.githubusercontent.com/8683/99021210-c8558e80-2525-11eb-8b80-f06426ecae11.gif)

At this point, given what I've seen so far, I'd expect command-} to switch me out of the whole nested Wallaby Jest tab into `cy:start`... or at least to go somewhere.

Worse, at this point command-{ doesn't do anything either, so I can't get back out the way I came in. I'd expect it to switch back to the Value Explorer, and I'd expect pressing it again to switch back to Wallaby Console, and so on.

The net result is that as I'm moving around from tab to tab, the Wallaby plugin is a black hole from which no keyboard navigation can escape (that I know of).

Interestingly, at this point – with Wallaby showing the Value Explorer | Debugger tab – if I first use the mouse to switch away from Wallaby, then Wallaby stops capturing the next/previous tab commands at all. That is, starting from `dev`, I can press command-} and command-{ all I want, and it just hops between Wallaby Jest and the other top-level tabs. (It would be okay with me if Wallaby _never_ captured these commands, and I had to use the mouse to switch between Wallaby subtabs, because I don't do as much switching within Wallaby. But I'd rather have it all navigable by keyboard.)

![2020-11-12 at 8 33 PM](https://user-images.githubusercontent.com/8683/99021639-b6c0b680-2526-11eb-9b8f-617a1dda4c86.gif)

Last data point: If Wallaby last showed Failing Tests, and I start from `cy:start` and press command-{ (moving left this time), I get:

1. Wallaby Jest, displaying Failing Tests, the leftmost tab.
2. Wrap around to Value Explorer | Debugger, the rightmost tab (showing the Debugger).
3. Nothing happens!

Getting stuck in the Debugger tab is the biggest problem, but the second step, where previous-tab wraps around to the rightmost inner tab, is also problematic. If step 3 were fixed, and step 2 wrapped around, I'd still have no way to switch out of Wallaby and back into my other processes.

![2020-11-12 at 8 38 PM](https://user-images.githubusercontent.com/8683/99021799-14550300-2527-11eb-9aa3-48a28bd0f19e.gif)

### Wallaby diagnostics report

I'm redacting most of the diagnostics report because the code I'm testing is proprietary and I don't think this behavior is tied to any of the tests that are run, but if you need a full report let me know and I'll set up a minimal repo I can send a report on.

```
{
  editorVersion: 'RubyMine 2020.2.3',
  pluginVersion: '1.0.202',
  editorType: 'IntelliJ',
  osVersion: 'darwin 19.6.0',
  nodeVersion: 'v14.4.0',
  coreVersion: '1.0.987',
  checksum: 'MTVkNGQ4YzRiZGUzZjE0YzBkODM1ZDBhZDRhMmUzNDcsMTYzMzczNzYwMDAwMCww',
  // ...
  debug: [
    // ...
    '2020-11-13T02:06:58.344Z workers Merging parallel test run results\n',
    '2020-11-13T02:06:58.363Z project Test run finished\n',
    '2020-11-13T02:06:58.363Z project Processed console.log entries\n',
    '2020-11-13T02:06:58.366Z project Processed loading sequences\n',
    '2020-11-13T02:06:58.373Z project Processed executed tests\n',
    '2020-11-13T02:06:58.423Z project Processed code coverage\n',
    '2020-11-13T02:06:58.610Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by smcenlly** [comment]
Thanks for raising the issue. We will investigate after the weekend.

**Comment by smcenlly** [comment]
Thanks for reporting the problem with the debugger tab, we were not aware that this was breaking keyboard support.

In terms of keyboard navigability, are you aware that there is a keyboard shortcut to show/navigate to an Active Tool Window? You can find the mapping for your keyboard layout by searching the keymap for `Show List of Tabs`:
![image](https://user-images.githubusercontent.com/2074089/99213037-4b256480-2818-11eb-9e17-a7649759ffe9.png)

The current behavior where IntelliJ next focused element gets "stuck" is a known JetBrains limitation and does not only affect Wallaby. The feature request to change this behavior is tracked here: [IDEA-164210](https://youtrack.jetbrains.com/issue/IDEA-164210?_ga=2.202859609.918802954.1605483534-1450887018.1559689100). Unfortunately there is nothing we can do to fix this on our end.

We have however found/fixed Wallaby's issue where the Debugger tab was ignore keyboard shortcuts (fixed in Wallaby for IntelliJ v1.0.203). You may use the `Show List of Tabs` keyboard shortcut to navigate with the keyboard out of Wallaby context (until/unless IntelliJ editors behavior changes).

**Comment by eostrom** [comment]
Thanks for investigating, and for the `Show List of Tabs` tip. There sure is a lot to know about JetBrains IDEs.
